### ISSUE TITLE
New feature .reattach()

### ISSUE THREAD

**Issue by Mr21** [jquery/jquery#2873]
Hi :)

Why not creating a `.reattach` methode?
So this: `jqElement.detach().reattach()` will change nothing.

It's will more easier to cancel a `.detach()` operation (eg. drag and drop).
And it can be use everytime like this:
``` javascript
jqCollection
    .detach()
    .css( { prop:values, ... } )
    .toggleClass( "a b c" )
    ...
    .reattach()
;
```
to accelerate the traitement, faster than `.css( "display", "none|block" )`.

To avoid any bug we can save the next sibling (to use the native `insertBefore`) **and** the parent in case if there was no next element or if it's deleted, to allow some usecase like:

``` javascript
jqCollection
    .detach()
    .parent()
        .empty()
        .end()
    .reattach()
;
```

**Comment by scottgonzalez** [comment]
I expect this would break very easily as collections go through transformations (e.g., `.map()`). Also, that code block isn't going to do *anything*. As soon as you detach the element(s), `.parent()` is going to return an empty set, so `.empty()` will not run on any elements.

**Comment by Mr21** [comment]
Oh... my bad, I've wrote this example without testing it...
``` javascript
jqParent = jqCollection.parent();
jqCollection.detach();
jqParent.empty();
jqCollection.reattach();
;
```
Thanks for reviewing this idea, I didn't understand the problem with `.map`, I've imagined we save all next and parent element for each element of the collection, not once for the all collection.

**Comment by timmywil** [comment]
What's the difference between `reattach` and appending other than remembering the parent?

**Comment by Mr21** [comment]
Curently, if we have to detach a collection and remembering where each element was, I suppose we have to write something like:
``` javascript
// detach:
jqCollection
	.each( function() {
		var jqThis = $( this );
		jqThis.data( {
			tmpNodeParent: jqThis.parent()[ 0 ],
			tmpNodeNext: jqThis.next()[ 0 ]
		})
	})
	.detach()
;

// reattach:
jqCollection.each( function() {
	var data = $.data( this ),
		jqThis = $( this );

	if ( data.tmpNodeNext ) {
		jqThis.insertBefore( data.tmpNodeNext );
	} else if ( data.tmpNodeParent ) {
		jqThis.appendTo( data.tmpNodeParent );
	}
});
```
And I was thinking maybe jQuery can propose a shortcut to do it faster.
There are two advantages to use something like `reattach`:
* Cancelling a drag.
* The possibility to accelerate some changes with: `jqCollection.detach()/*code*/.reattach();`

**Comment by timmywil** [comment]
Thanks for the request, but I think it is likely we'll introduce unexpected behaviors when it comes down to implementing this, and I'm not convinced that it's usefulness outweighs the cost. I'd prefer to let this start out as a plugin.
